kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableImpl.java

```diff
@@ -191,41 +184,15 @@ public CloseableIterator<ColumnarBatch> getChanges(
       long startVersion,
       long endVersion,
       Set<DeltaLogActionUtils.DeltaAction> actionSet) {
-    // Create a new action set which is a super set of the requested actions.
-    // The extra actions are needed either for checks or to extract
-    // extra information. We will strip out the extra actions before
-    // returning the result.
-    Set<DeltaLogActionUtils.DeltaAction> copySet = new HashSet<>(actionSet);
-    copySet.add(DeltaLogActionUtils.DeltaAction.PROTOCOL);
-    // commitInfo is needed to extract the inCommitTimestamp of delta files
-    copySet.add(DeltaLogActionUtils.DeltaAction.COMMITINFO);
-    // Determine whether the additional actions were in the original set.
-    boolean shouldDropProtocolColumn =
-        !actionSet.contains(DeltaLogActionUtils.DeltaAction.PROTOCOL);
-    boolean shouldDropCommitInfoColumn =
-        !actionSet.contains(DeltaLogActionUtils.DeltaAction.COMMITINFO);
 
-    return getRawChanges(engine, startVersion, endVersion, copySet)
-        .map(
-            batch -> {
-              int protocolIdx = batch.getSchema().indexOf("protocol"); // must exist
-              ColumnVector protocolVector = batch.getColumnVector(protocolIdx);
-              for (int rowId = 0; rowId < protocolVector.getSize(); rowId++) {
-                if (!protocolVector.isNullAt(rowId)) {
-                  Protocol protocol = Protocol.fromColumnVector(protocolVector, rowId);
-                  TableFeatures.validateKernelCanReadTheTable(protocol, getDataPath().toString());
-                }
-              }
-              ColumnarBatch batchToReturn = batch;
-              if (shouldDropProtocolColumn) {
-                batchToReturn = batchToReturn.withDeletedColumnAt(protocolIdx);
-              }
-              int commitInfoIdx = batchToReturn.getSchema().indexOf("commitInfo");
-              if (shouldDropCommitInfoColumn) {
-                batchToReturn = batchToReturn.withDeletedColumnAt(commitInfoIdx);
-              }
-              return batchToReturn;
-            });
+    logger.info(
+        "{}: Getting the commit files for versions [{}, {}]", tablePath, startVersion, endVersion);
```

Comment:
Can't this `logger.info` go into `DeltaLogActionUtils.getCommitFilesForVersionRange`

Reply:
i.e. I'm a fan of pushing logs like this into the underlying called methods.

Instead of 

```
// call site A
doLog
callMethod

// call site B
doLog
callMethod
```

we can just do

```
// call site A
callMethod

// call site B
callMethod

and then the underlying method does the log
```

Reply:
I agree -- just lost in the refactor. Updated.